src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1045,7 +1047,7 @@ def _validate_admin_password(password, os_type):
                              contains_digit, contains_special_char] if x])
     # pylint: disable=line-too-long
     if count < 3:
-        raise CLIError('Password must have the 3 of the following: 1 lower case character, 1 upper case character, 1 number and 1 special character')
+        raise CLIError("Your password is invalid for it violates Rule 2\n{}".format(error_msg))
```

Comment:
Maybe we can just show the rule that the user violates instead of printing all rules.

Reply:
That's exactly what the issue was open for. They want to know all the rules once they failed :(

Reply:
I suggest create a survey to collect users' opinions widely. That's just one user's appeal.

Reply:
Our PM opened this issue, I think she knows the users well ~